File: CrmApp.Client/src/models/ContractorHoursSnapshots.ts
import { UsersProfiles } from "./UsersProfiles";

export class ContractorHoursSnapshots {
    id: number = 0;
    contractorContractId: number = 0;
    snapshotDate: Date = new Date();
    hoursUsed: number = 0;
    hoursRemaining: number = 0;
    createdAt: Date = new Date();
    createdBy: number = 0;
    createdByUser?: UsersProfiles;
}

File: CrmApp.Domain/Entities/ContractorHoursSnapshots.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.Entities
{
    public class ContractorHoursSnapshots
    {
        [Key]
        public int Id { get; set; }
        public int ContractorContractId { get; set; }
        public DateTime SnapshotDate { get; set; }
        public decimal HoursUsed { get; set; }
        public decimal HoursRemaining { get; set; }
        public DateTime CreatedAt { get; set; }
        public int CreatedBy { get; set; }
        [ForeignKey(nameof(ContractorContractId))]
        public virtual ContractorContracts? ContractorContract { get; set; }
        [ForeignKey(nameof(CreatedBy))]
        public virtual UsersProfiles? CreatedByUser { get; set; }
    }
}
